gradio/blocks.py

```diff
@@ -228,6 +229,8 @@ def __init__(
         self.is_space = True if os.getenv("SYSTEM") == "spaces" else False
         self.favicon_path = None
         self.auth = None
+        self.dev_mode = True
```

Comment:
why True by default?

Reply:
The reason is that `uvicorn` bypasses the standard `launch()` method. So if you're using `uvicorn` to launch in debug mode, then `dev_mode` stays True, otherwise changes to `False`.